doc/source/user/tutorial-svd.rst

```diff
@@ -0,0 +1,494 @@
+
+::
+
+    >>> Sigma = np.zeros((3, 768, 1024))
+    >>> for j in range(3):
+    ...     np.fill_diagonal(Sigma[j, :, :], s[j, :])
+
+Now, if we wish to rebuild the full SVD (with no approximation), we can do
+
+::
+
+    >>> reconstructed = U @ Sigma @ Vt
+
+Note that
+
+::
+
+    >>> reconstructed.shape
+    (3, 768, 1024)
+
+and
+
+::
+
+    >>> plt.imshow(np.transpose(reconstructed, (1, 2, 0)))
+
+.. plot:: user/plot_reconstructed.py
+    :align: center
+    :include-source: 0
+    
+should give you an image indistinguishable from the original one (although we
+may introduce floating point errors for this reconstruction. In fact, 
+you might see a warning message saying `"Clipping input data to the
+valid range for imshow with RGB data ([0..1] for floats or [0..255] for
+integers)."` This is expected from the manipulation we just did on the original
+image.
+
+Now, to do the approximation, we must choose only the first ``k`` singular
+values for each color channel. This can be done using the following syntax::
+
+    >>> approx_img = U @ Sigma[..., :k] @ Vt[..., :k, :]
+
+You can see that we have selected only the first ``k`` components of the last
+axis for ``Sigma`` (this means that we have used only the first ``k`` columns
+of each of the three matrices in the stack), and that we have selected only the
+first ``k`` components in the second-to-last axis of ``Vt`` (this means we have
+selected only the first ``k`` rows from every matrix in the stack ``Vt`` and
+all columns). If you are unfamiliar with the ellipsis syntax, it is a
+placeholder for other axes. For more details, see the documentation on
+:ref:`Indexing <basics.indexing>`.
+
+Now,
+
+::
+
+    >>> approx_img.shape
+    (3, 768, 1024)
+
+which is not the right shape for showing the image. Finally, reordering the
+axes back to our original shape of ``(768, 1024, 3)``, we can see our approximation::
+
+    >>> plt.imshow(np.transpose(approx_img, (1, 2, 0)))
+
+.. plot:: user/plot_final.py
+    :align: center
+    :include-source: 0
+    
+Even though the image is not as sharp, using a small number of ``k`` singular
+values (compared to the original set of 768 values), we can recover many of the
+distinguishing features from this image.
+
+**Final words**
+
+Of course, this is not the best method to *approximate* an image.
+However, there is, in fact, a result in linear algebra that says that the
+approximation we built above is the best we can get to the original matrix in
+terms of the norm of the difference. For more information, see *G. H. Golub and
+C. F. Van Loan, Matrix Computations, Baltimore, MD, Johns Hopkins University
+Press, 1985*.
+
+**Further reading**
+
+-  `Python tutorial <https://docs.python.org/tutorial/>`__
+-  :ref:`reference`
+-  `SciPy Tutorial <https://docs.scipy.org/doc/scipy/reference/tutorial/index.html>`__
+-  `SciPy Lecture Notes <https://scipy-lectures.org>`__
+-  A `matlab, R, IDL, NumPy/SciPy dictionary <http://mathesaurus.sf.net/>`__
```

Comment:
The first three could become (scipy lecture notes' ref is `index`, that seems too generic to be useful)
```
- :ref:`Python Tutorial <tutorial-index>`
- :ref:`reference`
- :ref:`SciPy Tutorial <tutorial/index>` 
```

Reply:
I did the same as with the links above. Thanks!